crates/ibc/src/applications/transfer/denom.rs

```diff
@@ -147,10 +149,11 @@ impl Display for TracePath {
 }
 
 /// A type that contains the base denomination for ICS20 and the source tracing information path.
-#[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
 pub struct PrefixedDenom {
     /// A series of `{port-id}/{channel-id}`s for tracing the source of the token.
-    #[serde(with = "serde_string")]
+    #[cfg_attr(feature = "serde", serde(with = "serde_string"))]
```

Comment:
Why is the `serde_string` module necessary? Also the serialization/deserialization is based on the `Display` implementation, which seems brittle and easy to break

Reply:
The `Coin::amount` is serialized as a decimal string so we use the display impl for easy serde. I think this is a popular pattern and there are crates that provide this functionality e.g. [serde_with](https://docs.rs/serde_with/2.2.0/serde_with/struct.DisplayFromStr.html). Can you please expand on why you think this is brittle?

Reply:
I find it brittle because in general, `Display` is for human-readable output; it's easy to forget that it's also used for serialization. If we change it for a different output, we just made a breaking serialization change, which is pretty bad. I find it unfortunate that it's a common pattern... We can leave it as is though